met-web/src/components/engagement/view/EmailModal.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+import { Modal } from '@mui/material';
+import SuccessPanel from './SuccessPanel';
+import FailurePanel from './FailurePanel';
+import EmailPanel from './EmailPanel';
+import TabContext from '@mui/lab/TabContext';
+import TabPanel from '@mui/lab/TabPanel';
+import { EmailModalProps } from './types';
+import { checkEmail } from 'utils';
+
+function EmailModal(props: EmailModalProps) {
+    const [formIndex, setFormIndex] = useState('email');
+    const [email, setEmail] = useState('');
+
+    function handleClose() {
+        props.handleClose();
+        setFormIndex('email');
+    }
+
+    function updateTabValue() {
+        if (!checkEmail(email)) {
+            setFormIndex('error');
+        } else {
+            setFormIndex('success');
+        }
+    }
+
+    return (
+        <>
```

Comment:
no need for the Fragments <></> the Modal component already contains everything. <></> are usually useful when we don't have one container containing all children components